### ISSUE TITLE
Disagreement on “keep it short”, “make it easy to type”

### ISSUE THREAD

**Issue by willghatch** [cli-guidelines/cli-guidelines#35]
I think the advice in the “keep it short” and “make it easy to type” sections is misguided, aside from the sage advice to “not make it too short”.  The shortest combinations should in fact not be reserved for “common utilities used all the time” like `ls`, `ps`, etc, but for user-made aliases and scripts.  By this document's own admission these commands are outdated and have a poor interface.  Every name chosen that scripts or programs come to rely upon becomes a name that the user can't put on `PATH` as a script, meaning that the user has to find some other name or risk strange breakage.  Program authors shouldn't assume that their utility is commonly used.  For most users, `gs` is likely much better as an alias for `git status` than as the name for the `ghostscript` executable.  Users who want to use the name `ls` for file listing have to choose between the poor UI of the traditional `ls` command, having inconsistent behavior by using a shell alias (eg. shell aliases don't work with `find -exec` and similar commands), or making a wrapper script that breaks every other program that relies on the semantics of traditional unix `ls`.  Every command name chosen is essentially taking a piece of a global namespace that users have to work around for names they choose for *their* most commonly used commands.  Keep names unique, descriptive, and sufficiently long.  If a user uses a given command enough that typing it is a burden, they can create a short-named alias or wrapper script.

Also, optimizing names for the qwerty keyboard is silly.  Use good names because they are good names, not because a keyboard layout designed for 1800s technological constraints and goals is still (mis)used today in a mostly unrelated context.

**Comment by Quelklef** [comment]
I agree with your first paragraph, but I somewhat disagree with your second one. The QWERTY layout is arbitrary, but we're also stuck with it, at least for now. And so if you're picking between two comparably-good names, it makes sense to use ease-of-typing as a metric to help you decide one way or another. (Another worthwhile metric is ease-of-reading, e.g. `Odyssey` vs `Illiad`)
